lib/game.rb

```diff
@@ -31,36 +34,51 @@ def initialize(
         )
         @status = :play
         @board_presenter = BoardPresenter.new(presenter: board_presenter)
+        @repository_type = repository_type
     end
 
     def play
         while status == :play
             output_board
-            position = get_valid_move(board: game_state.board, presenter: game_io)
-            board = update_board(input: position)
-            update_game_status(board: board)
-            game_state.switch_players if status == :play
+            output_save_option
+            input = valid_input(board: game_state.board, presenter: game_io)
+            if input == InputType::SAVE
+                @status = :save
+            else
+                board = update_board(input: input) 
+                update_game_status(board: board)
+                game_state.switch_players if status == :play
+            end
         end
         output_board
+        save_game if status == :save
         output_win if status == :win
```

Comment:
Similar comment here about the branching logic - may make sense to change this to an `if/elsif` since it would never be both a win and a tie.

Reply:
would if / else be better than just doing an explicit return?
```
   return save_game if status == :save
  return output_win if status == :win
    return output_tie if status == :tie 
```